src/rabbit.erl

```diff
@@ -1236,6 +1271,112 @@ validate_msg_store_io_batch_size_and_credit_disc_bound(CreditDiscBound,
             end
     end.
 
+-spec product_name() -> string().
+
+product_name() ->
+    #{name := ProductName} = product_infos(),
+    ProductName.
+
+-spec product_version() -> string().
+
+product_version() ->
+    #{version := ProductVersion} = product_infos(),
+    ProductVersion.
+
+-spec product_infos() -> #{name := string(),
+                           version := string()}.
+
+product_infos() ->
+    PTKey = {?MODULE, product},
+    try
+        %% The value is cached the first time to avoid calling the
+        %% application master many times just for that.
+        persistent_term:get(PTKey)
+    catch
+        error:badarg ->
+            %% FIXME: Move those variables to rabbit_env.
```

Comment:
I cannot think of a good reason to allow this to be configured via an environment variable. I am leaning towards keeping this as application config only.

What was your thinking behind this?

Reply:
I generally like the ability to set things through the environment. This made manual testing way easier for instance.

Reply:
I'm not saying these variables should be documented and people should rely on them though.

Reply:
I see. If we agree to not document this and only use it for testing, happy to keep as is.